All_Relationships/2.data-labeler.py

```diff
@@ -0,0 +1,221 @@
+
+# coding: utf-8
+
+# # MUST RUN AT THE START OF EVERYTHING
+
+# Load all the imports and set up the database for database operations. Plus, set up the particular candidate type this notebook is going to work with. 
+
+# In[2]:
+
+
+get_ipython().magic(u'load_ext autoreload')
+get_ipython().magic(u'autoreload 2')
+get_ipython().magic(u'matplotlib inline')
+
+from collections import defaultdict
+import re
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import tqdm
+
+
+# In[ ]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[ ]:
+
+
+from snorkel.annotations import FeatureAnnotator, LabelAnnotator
+from snorkel.features import get_span_feats
+from snorkel.models import candidate_subclass
+from snorkel.models import Candidate
+from snorkel.viewer import SentenceNgramViewer
+
+
+# In[ ]:
+
+
+edge_type = "dg"
+debug = False
+
+
+# In[ ]:
+
+
+if edge_type == "dg":
+    DiseaseGene = candidate_subclass('DiseaseGene', ['Disease', 'Gene'])
+    edge = "disease_gene"
+elif edge_type == "gg":
+    GeneGene = candidate_subclass('GeneGene', ['Gene1', 'Gene2'])
+    edge = "gene_gene"
+elif edge_type == "cg":
+    CompoundGene = candidate_subclass('CompoundGene', ['Compound', 'Gene'])
+    edge = "compound_gene"
+elif edge_type == "cd":
+    CompoundDisease = candidate_subclass('CompoundDisease', ['Compound', 'Disease'])
+    edge = "compound_disease"
+else:
+    print("Please pick a valid edge type")
+
+
+# # Look at potential Candidates
+
+# Use this to look at loaded candidates from a given set. The constants represent the index to retrieve the appropiate set. Ideally, here is where one can look at a subset of the candidate and develop label functions for candidate labeling.
+
+# In[ ]:
+
+
+TRAIN = 0
+DEV = 1
+
+
+# In[ ]:
+
+
+candidates = session.query(DiseaseGene).filter(DiseaseGene.split==TRAIN).limit(100)
+sv = SentenceNgramViewer(candidates, session)
+
+
+# In[ ]:
+
+
+sv
+
+
+# # Label Functions
+
+# Here is one of the fundamental part of this project. Below are the label functions that are used to give a candidate a label of 1,0 or -1 which corresponds to correct label, unknown label and incorrection label. The goal here is to develop functions that can label accurately label as many candidates as possible. This idea comes from the [data programming paradigm](https://papers.nips.cc/paper/6523-data-programming-creating-large-training-sets-quickly), where the goal is to be able to create labels that machine learning algorithms can use for accurate classification.  
+
+# In[ ]:
+
+
+if edge_type == "dg":
+    from utils.disease_gene_lf import *
```

Comment:
best to avoid `import *`, which makes it difficult for readers to know where imports are coming from. Would it be problematic to just list the functions being imported?